### PR TITLE
Attempt to skip zero bytes at the end of Scan blocks when decoding JPEG images (issue 4090)

### LINKED ISSUE #4090: JPEG error: marker was not found
https://www.ets.org/Media/Tests/GRE/pdf/gre_research_validity_data.pdf

```
"PDF d0331af6364b4637a61316757c264737 [1.6 PDFlib Personalization Server 7.0.0p3 (C++/Win32) / Rampage PDF PageAssembly] (PDF.js: 0.8.827)" viewer.js:2897
"Error: JPEG error: marker was not found" pdf.worker.js:216
"JpegStream_ensureBuffer@resource://pdf.js/build/pdf.worker.js:32948\nDecodeStream_getBytes@resource://pdf.js/build/pdf.worker.js:32246\nPDFImage_getImageBytes@resource://pdf.js/build/pdf.worker.js:28437\nPDFImage_fillRgbaBuffer@resource://pdf.js/build/pdf.worker.js:28369\nPDFImage_getImageData@resource://pdf.js/build/pdf.worker.js:28432\nPartialEvaluator_buildPaintImageXObject/<@resource://pdf.js/build/pdf.worker.js:14618\nPDFImage_buildImage/<@resource://pdf.js/build/pdf.worker.js:28085\n" pdf.worker.js:218
"Warning: Unsupported feature \"unknown\""
```

### PR DESCRIPTION
**Pull request by Snuffleupagus**
Fixes #4090.

### CODE REVIEW

File: src/core/jpg.js [hunk 1]
```diff
@@ -363,6 +363,12 @@ var JpegImage = (function jpegImage() {
       // find marker
       bitsCount = 0;
       marker = (data[offset] << 8) | data[offset + 1];
+      // Some bad images seem to pad Scan blocks with zero bytes, skip past
+      // those to attempt to find a valid marker (fixes issue4090.pdf).
+      while (marker === 0xFF && offset < data.length - 1) {
```
Comment (yurydelendik) [inline src/core/jpg.js]: I have a feeling that `marker === 0xFF` is not valid comparison here. It looks like the loop only skips one zero byte. Shall we make this an `if`?

Comment (Snuffleupagus) [inline src/core/jpg.js]: How about this instead, to better align the code with the comment and also to support multiple zero bytes?
```javascript
while (data[offset] === 0x00 && offset < data.length - 1) {
  ...
```
Or, I suppose that we could just do:
```javascript
if (data[offset] === 0x00 && offset < data.length - 1) {
  ...
```
Which one do you prefer?

Comment (yurydelendik) [inline src/core/jpg.js]: First will work.

**Review by yurydelendik — APPROVED** [review]
Thank you for the patch.

**Comment by yurydelendik** [comment]
Thank you for the patch

**Pull request merged**